reliquery/storage.py

```diff
@@ -425,15 +366,145 @@ def get_all_relic_tags(self) -> List[Dict]:
+                        }
+                    )
+            return relic_data
+
+
+class DropboxStorage(Storage):
+    def __init__(
+        self,
+        access_token: str,
+        prefix: str,
+        name: str,
+    ):
+        self.dbx = dropbox.Dropbox(access_token)
+        self.prefix = "/" + prefix
+        self.name = name
+
+    def _join_path(self, path: StoragePath) -> str:
+        return "/".join([self.prefix] + path)
+
+    def put_file(self, path: StoragePath, file_path: str) -> None:
+        path = path.split()
+        with open(file_path, "rb") as f:
+            self.dbx.files_upload(f.read(), "/" + self._join_path(path), mute=True)
+
+    def put_binary_obj(self, path: StoragePath, buffer: BytesIO) -> None:
+        self.dbx.files_upload(buffer.read(), self._join_path(path), mute=True)
+
+    def get_binary_obj(self, path: StoragePath) -> BytesIO:
+        return BytesIO(
+            self.dbx.files_download(self._join_path(path), rev=None)[-1].content
+        )
+
+    def put_text(self, path: StoragePath, text: str, encoding: str = "utf-8") -> None:
+        string_bytes = bytes(text, encoding)
+        self.dbx.files_upload(string_bytes, self._join_path(path))
+
+    def get_text(self, path: StoragePath) -> str:
+        try:
+            return self.dbx.files_download(self._join_path(path))[-1].text
+        except ApiError:
+            raise StorageItemDoesNotExist
+
+    def list_keys(self, path: StoragePath) -> List[str]:
+        try:
+            return [
+                i.name
+                for i in self.dbx.files_list_folder(self._join_path(path)).entries
+            ]
+        except ApiError:
+            return []
+
+    def put_metadata(self, path: StoragePath, metadata: Dict):
+        metadata_bytes = json.dumps(metadata).encode("utf-8")
+        self.dbx.files_upload(
+            metadata_bytes,
+            self._join_path(path),
+            mode=dropbox.files.WriteMode.overwrite,
+        )
+
+    def get_metadata(self, path: StoragePath, root_key: str) -> Dict:
+        data = {root_key: {k: [] for k in DATA_TYPES}}
+
+        def dict_from_path(path: StoragePath, dirname: str):
+            dirpath = path.copy()
+            dirpath.append(dirname)
+            entries = self.list_keys(dirpath)
+
+            for i in entries:
+                entry_path = dirpath.copy()
+                entry_path.append(i)
+                data[root_key][dirname].append(
+                    self.dbx.files_download(self._join_path(entry_path))[-1].json()
+                )
+
+        for d in data[root_key]:
+            dict_from_path(path, d)
+
+        return data
+
+    def list_key_paths(self, path: StoragePath) -> List[str]:
+        paths = []
+        subs = self.list_keys(path)
+        if subs:
+            for sub in subs:
+                copy = path.copy()
+                copy.append(sub)
+                sub_path = self._join_path(copy)
+                if isinstance(
```

Comment:
I think we may want to add a clarifying comment here that using isntstance was the best way to do this. (just so future readers of the code understand why isinstance was used).

Reply:
isinstace -> isinstance